### PR TITLE
Survival learner doc updates + miscellaneous stuff

### PR DESCRIPTION
**Pull request by bblodfon**
# Overview of changes
- Add prediction types `@section` for all `survival` learners, making sure it is well documented and consistent #347 
- All survival learners have `crank` as main prediction (and it is always returned) #331 
- Other stuff: 
  - Set `n_thread = 1` for `surv.aorsf` and use unique event time points for predicted `S(t)` @bcjaeger  
  - Add `selected_features()` for `surv.penalized`
  - Fix `surv.prioritylasso` learner + add `distr` predictions via Breslow #344
  - SSVM `gamma.mu` was splitted to `gamma` and `mu`
  - Harmonized the use of times points for prediction as much as possible across survival learners #387 
    - `parametric` and `akritas` get a proper `ntime` argument
  - `parametric` is now used by default with `discrete = TRUE`

# List of survival learners (WIP)
- aorsf, BART, 2 x coxboost, survflex, gbm, glmnet x 2, mboost x 4 {gamboost, glmboost, mboost, blackboost}, cforest, ctree, penalized, prioritylasso, survival SVM, rfsrc, ranger, akritas, parametric, nelson, 6 x DL survival models (coxtime, deepsurv, loghaz, deephit, dnnsurv, pchazard), 2 x xgboost (cox, aft)

### CODE REVIEW

File: R/learner_aorsf_surv_aorsf.R [hunk 1]
```diff
@@ -45,7 +46,7 @@ LearnerSurvAorsf = R6Class("LearnerSurvAorsf",
         n_tree = p_int(default = 500L, lower = 1L, tags = "train"),
         n_split = p_int(default = 5L, lower = 1L, tags = "train"),
         n_retry = p_int(default = 3L, lower = 0L, tags = "train"),
-        n_thread = p_int(default = 0, lower = 0, tags = c("train", "predict")),
+        n_thread = p_int(default = 0, lower = 0, tags = c("train", "predict", "nthreads")),
```
Comment (sebffischer) [inline R/learner_aorsf_surv_aorsf.R:49]:
```suggestion
        n_thread = p_int(default = 0, lower = 0, tags = c("train", "predict", "threads")),
```

File: R/learner_partykit_surv_cforest.R [hunk 2]
```diff
@@ -112,11 +114,14 @@ LearnerSurvCForest = R6Class("LearnerSurvCForest",
       ps$values$mincriterion = 0
       ps$values$saveinfo = FALSE
 
+      # set cores to 1
+      ps$values$cores = 1
```
Comment (sebffischer) [inline R/learner_partykit_surv_cforest.R:118]: it's now also possible to do `p_int(..., init = 1)` but this is also fine

File: R/learner_flexsurv_surv_flexible.R [hunk 3]
```diff
@@ -111,8 +111,8 @@ predict_flexsurvreg = function(object, task, learner, ...) {
   }
 
   X = stats::model.matrix(formulate(rhs = task$feature_names),
-    data = newdata,
-    xlev = task$levels())
+                          data = newdata,
```
Comment (sebffischer) [inline R/learner_flexsurv_surv_flexible.R:114]: I am not too fond of this type of indentation style as it wastes a lot of space for nothing

File: R/learner_prioritylasso_regr_priority_lasso.R [hunk 4]
```diff
@@ -6,6 +6,9 @@
 #' Patient outcome prediction based on multi-omics data taking practitioners’ preferences into account.
 #' Calls [prioritylasso::prioritylasso()] from \CRANpkg{prioritylasso}.
 #'
+#' @section Custom mlr3 parameters:
```
Comment (sebffischer) [inline R/learner_prioritylasso_regr_priority_lasso.R:9]: are these initial values or custom paramters? (custom parameters are those not present in the upstream learner like mtry.ratio for ranger)

File: R/learner_prioritylasso_surv_priority_lasso.R [hunk 5]
```diff
@@ -5,6 +5,25 @@
...
+#' 1. `lp`: a vector containing the linear predictors (relative risk scores),
+#' where each score corresponds to a specific test observation.
+#' Calculated using [prioritylasso::predict.prioritylasso()].
+#' 2. `crank`: same as `lp`.
+#' 3. `distr`: a survival matrix in two dimensions, where observations are
+#' represented in rows and time points in columns.
+#' Calculated using [mlr3proba::breslow()] where the Breslow estimator is used
+#' for computing the baseline hazard.
+#'
+#' @section Custom mlr3 parameters:
```
Comment (sebffischer) [inline R/learner_prioritylasso_surv_priority_lasso.R:24]: Again I think these are initial values and not custom parameters I assume?

File: R/learner_randomForestSRC_classif_imbalanced_rfsrc.R [hunk 6]
```diff
@@ -16,7 +16,28 @@
 #' `r format_bib("obrien2019imbrfsrc", "chen2004imbrf")`
 #'
 #' @template seealso_learner
-#' @template example
+#' @examples
```
Comment (sebffischer) [inline R/learner_randomForestSRC_classif_imbalanced_rfsrc.R:19]: example needs to be escaped if the packages it not available

File: R/learner_randomForestSRC_classif_imbalanced_rfsrc.R [hunk 7]
```diff
@@ -16,7 +16,28 @@
...
+#' # Define the Learner
+#' learner = mlr3::lrn("classif.imbalanced_rfsrc", importance = "TRUE")
```
Comment (sebffischer) [inline R/learner_randomForestSRC_classif_imbalanced_rfsrc.R:21]: why TRUE in quotes?

File: R/learner_randomForestSRC_classif_rfsrc.R [hunk 8]
```diff
@@ -10,22 +10,40 @@
...
-#'     Note that `sampsize` and `sampsize.ratio` are mutually exclusive.
-#' - `cores`:
-#'     This value is set as the option `rf.cores` during training and is set to 1 by default.
+#' - `mtry`: This hyperparameter can alternatively be set via the added
+#'  hyperparameter `mtry.ratio` as `mtry = max(ceiling(mtry.ratio * n_features), 1)`.
+#'  Note that `mtry` and `mtry.ratio` are mutually exclusive.
+#' - `sampsize`: This hyperparameter can alternatively be set via the added
+#'  hyperparameter `sampsize.ratio` as `sampsize = max(ceiling(sampsize.ratio * n_obs), 1)`.
+#'  Note that `sampsize` and `sampsize.ratio` are mutually exclusive.
+#' - `cores`: This value is set as the option `rf.cores` during training and is
```
Comment (sebffischer) [inline R/learner_randomForestSRC_classif_rfsrc.R:19]: custom parameter or initial value?

File: R/learner_randomForestSRC_classif_rfsrc.R [hunk 9]
```diff
@@ -10,22 +10,40 @@
...
+#' set to 1 by default.
 #'
 #' @references
 #' `r format_bib("breiman_2001")`
 #'
 #' @template seealso_learner
-#' @template example
+#' @examples
+#' # Define the Learner
```
Comment (sebffischer) [inline R/learner_randomForestSRC_classif_rfsrc.R:27]: escape example

File: R/learner_randomForestSRC_regr_rfsrc.R [hunk 10]
```diff
@@ -15,7 +15,27 @@
 #' `r format_bib("breiman_2001")`
 #'
 #' @template seealso_learner
-#' @template example
+#' @examples
```
Comment (sebffischer) [inline R/learner_randomForestSRC_regr_rfsrc.R:18]: escape

File: R/learner_randomForestSRC_classif_rfsrc.R (hunk 9 above)
Comment (sebffischer) [reply R/learner_randomForestSRC_classif_rfsrc.R:27]: Please check this everywhere

File: R/learner_survivalmodels_surv_parametric.R [hunk 11]
```diff
@@ -128,13 +135,22 @@ LearnerSurvParametric = R6Class("LearnerSurvParametric",
 
     .predict = function(task) {
       pv = self$param_set$get_values(tags = "predict")
-
       newdata = as.data.frame(ordered_features(task, self))
 
+      # use unique train set times
```
Comment (sebffischer) [inline R/learner_survivalmodels_surv_parametric.R:140]: This is used multiple times. Refactor this into a function

File: R/learner_survivalsvm_surv_svm.R [hunk 12]
```diff
@@ -18,14 +25,19 @@
 #'
 #' `makediff3` is recommended when using `type = "hybrid"`.
 #'
+#' @section Custom mlr3 parameters:
+#'
+#' - `gamma`, `mu` have replaced `gamma.mu` so that it's easier to tune these separately.
```
Comment (sebffischer) [inline R/learner_survivalsvm_surv_svm.R:30]: just fyi: This could have easily been achieved with a parameter trafo as well

File: R/learner_survivalsvm_surv_svm.R [hunk 13]
```diff
@@ -89,37 +102,55 @@ LearnerSurvSVM = R6Class("LearnerSurvSVM",
...
+          .args = pars)
       })
     },
 
     .predict = function(task) {
       pars = self$param_set$get_values(tags = "predict")
-      fit = predict(self$model, newdata = ordered_features(task, self),
-        .args = pars
-      )
+      fit = invoke(predict,
```
Comment (sebffischer) [inline R/learner_survivalsvm_surv_svm.R:139]: why are there so many indentation changes?
In general we want to change code as little as possible in a meaningful way so the diffs between files actually express relevant changes and not just style changes

File: R/learner_xgboost_surv_xgboost_aft.R [hunk 14]
```diff
@@ -26,7 +26,6 @@
 #' @template section_early_stopping
 #' @templateVar id surv.xgboost.aft
 #' @template learner
-#' @template section_early_stopping
```
Comment (sebffischer) [inline R/learner_xgboost_surv_xgboost_aft.R:29]: why was this removed?

**Review by sebffischer — COMMENTED** [review]
looking great, just some minor comments / questions

File: R/learner_partykit_surv_cforest.R (hunk 2 above)
Comment (bblodfon) [reply R/learner_partykit_surv_cforest.R:118]: ha, new syntax, nice! to be consistent with eveyr other learner for now let's leave it as is?

File: R/learner_randomForestSRC_classif_imbalanced_rfsrc.R (hunk 7 above)
Comment (bblodfon) [reply R/learner_randomForestSRC_classif_imbalanced_rfsrc.R:21]: the param doesn't take the logical `TRUE`! See error:
```
Error in self$assert(xs, sanitize = TRUE) : 
  Assertion on 'xs' failed: importance: Must be element of set {'FALSE','TRUE','none','permute','random','anti'}, but types do not match (logical != character).
```

File: R/learner_randomForestSRC_classif_imbalanced_rfsrc.R (hunk 6 above)
Comment (bblodfon) [reply R/learner_randomForestSRC_classif_imbalanced_rfsrc.R:19]: nice catch, I updated all examples from the learners I changed

File: R/learner_survivalsvm_surv_svm.R (hunk 12 above)
Comment (bblodfon) [reply R/learner_survivalsvm_surv_svm.R:30]: We had some much trouble with this doing it with the trafo, Lukas had made an issue about it => https://github.com/slds-lmu/paper_2023_survival_benchmark/issues/9

File: R/learner_survivalsvm_surv_svm.R (hunk 13 above)
Comment (bblodfon) [reply R/learner_survivalsvm_surv_svm.R:139]: haha yes - my feeling at the time was since I am looking/fixing/documenting at all these learners, I tried to harmonize also a bit the code, and well as you saw, I have my favorite style :) I will be a bit more reluctant to do this in the future

File: R/learner_survivalmodels_surv_parametric.R (hunk 11 above)
Comment (bblodfon) [reply R/learner_survivalmodels_surv_parametric.R]: Done!

**Review by sebffischer — APPROVED** [review]
